### PR TITLE
Adds webp width/height parsing for all three webp variants

### PR DESCRIPTION
**Pull request by d-Pixie**
Adds a pure nim, no external dependencies implementation to parse width/height from webp images.

### CODE REVIEW

File: src/image_formats/webp.nim [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import std/[streams,bitops,strutils,endians]
+
+
+proc get_size_from_vp8_header_block( stream: var FileStream ): (int,int)
+proc get_size_from_vp8l_header_block( stream: var FileStream ): (int,int)
+proc get_size_from_vp8_header_block( stream: var FileStream ): (int,int)
```
Comment (ehannes) [inline src/image_formats/webp.nim:6]: I don't know nim yet, but isn't this a duplicate of row 4? Do we really need it? :thinking:

File: src/image_formats/webp.nim [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+proc get_size*( path:string ): (int,int) =
+  var stream = new_file_stream( path, fm_read )
+  defer: stream.close()
+
+  stream.set_position(12)
```
Comment (ehannes) [inline src/image_formats/webp.nim:12]: Can we do something with these magic numbers? There are a bunch of them in this pull request.

File: src/image_formats/webp.nim [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+proc get_size_from_vp8_header_block( stream: var FileStream ): (int,int) =
+    let w = stream.read_extended_size_value_at_position( 24 )
+    let h = stream.read_extended_size_value_at_position( 27 )
+
+    return (int(w)+1,int(h)+1)
+
+
+
+assert get_size("examples/animated.webp") == (400,400)
```
Comment (ehannes) [inline src/image_formats/webp.nim:77]: Is this some sort of build in tests?

File: src/image_formats/webp.nim (hunk 2 above)
Comment (d-Pixie) [reply src/image_formats/webp.nim:12]: Open to suggestions 🙂

**Pull request merged**